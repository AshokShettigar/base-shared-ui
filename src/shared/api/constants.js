export const partnerId = '0';

// https://continuum.atlassian.net/wiki/spaces/C2E/pages/153880042/REST+API+Design+Guide
export const HTTP_STATUSES = {
    OK: 200,
    CREATED: 201,
    NO_CONTENT: 204,
    BAD_REQUEST: 400,
    UNAUTHORIZED: 401,
    FORBIDDEN: 403,
    NOT_FOUND: 404,
    METHOD_NOT_ALLOWED: 405,
    CONFLICT: 409,
    UNSUPPORTED_MEDIA_TYPE: 415,
    TOO_MANY_REQUESTS: 429,
    CLIENT_CLOSED_REQUEST: 499,
    INTERNAL_SERVER_ERROR: 500,
    SERVICE_UNAVAILABLE: 503
};

export const URLS = {
    INTEGRATION: 'https://integration.control.v2.itsupport247.net',
    LOG_OUT: 'https://control.dtitsupport247.net/logout.asp'
};
