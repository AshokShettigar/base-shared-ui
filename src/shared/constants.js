export const SIZES = {
    MINI: 'mini',
    TINY: 'tiny',
    SMALL: 'small',
    MEDIUM: 'medium',
    LARGE: 'large',
    BIG: 'big',
    HUGE: 'huge',
    MASSIVE: 'massive',
    BANNER: 'banner'
};

export const KINDS = {
    PRIMARY: 'primary',
    SECONDARY: 'secondary'
};
