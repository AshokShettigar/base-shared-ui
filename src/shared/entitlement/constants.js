export default {
    TASKING_BASIC: 'TASKING_BASIC',
    SEQUENCE_BASIC: 'SEQUENCE_BASIC',
    TASKING_DYNAMIC_GROUPS: 'TASKING_DYNAMIC_GROUPS',
    TASKING_TASKS_HOME_PAGE: 'TASKING_TASKS_HOME_PAGE',
    PATCHING_DESKTOPS: 'Desktops Patching UI',
    PATCHING_APP_FRESH_INSTALL: 'AppFreshInstall',
    PATCHING_DEVICE_LEVEL_ASSIGNMENT: 'PatchingDeviceLevelAssignment'
};
