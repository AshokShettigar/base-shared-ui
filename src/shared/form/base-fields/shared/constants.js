export const DAYS_OF_WEEK = [{
    shortName: 'Sun',
    name: 'Sunday',
    number: 0
}, {
    shortName: 'Mon',
    name: 'Monday',
    number: 1
}, {
    shortName: 'Tue',
    name: 'Tuesday',
    number: 2
}, {
    shortName: 'Wed',
    name: 'Wednesday',
    number: 3
}, {
    shortName: 'Thu',
    name: 'Thursday',
    number: 4
}, {
    shortName: 'Fri',
    name: 'Friday',
    number: 5
}, {
    shortName: 'Sat',
    name: 'Saturday',
    number: 6
}];
