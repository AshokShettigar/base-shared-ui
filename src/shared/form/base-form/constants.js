export const SELECTED_FIELD = 'SELECTED_FIELD';
