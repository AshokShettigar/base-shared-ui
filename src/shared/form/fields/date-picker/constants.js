export const defaultDateFormat = 'DD/MM/YYYY';
export const defaultTimeFormat = 'hh:mm A';

export const defaultDatePlaceholder = defaultDateFormat;
export const defaultTimePlaceholder = 'HH:MM';

export const VARIATION = {
    TIME: 'time',
    DATE: 'date',
    DATE_TIME: 'dateTime'
};
