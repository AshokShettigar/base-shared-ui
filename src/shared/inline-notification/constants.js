export const NOTIFICATION_ICON_TYPES = {
    info: 'info',
    warning: 'alert',
    error: 'remove',
    success: 'success',
    message: 'message'
};

export const ROOT_CLASS = 'inline-notification';
