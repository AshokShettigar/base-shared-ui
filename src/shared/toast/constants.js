const ROOT_CLASS = 'toast';
export default ROOT_CLASS;

export const TOAST_TYPES = {
    MESSAGE: 'message',
    SUCCESS: 'success',
    INFO: 'info',
    WARNING: 'warning',
    ERROR: 'error'
};
