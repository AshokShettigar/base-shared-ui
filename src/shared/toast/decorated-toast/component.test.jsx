import { shallow } from 'enzyme';
import DecoratedToast from './component';
import { TOAST_TYPES } from '../constants';

const defaultProps = {
    message: 'Toast Message',
    onClose: jest.fn(),
    type: TOAST_TYPES.MESSAGE
};

describe('Decorated Toast', () => {
    it('default renders', () => {
        const wrapper = shallow(<DecoratedToast {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders with props title', () => {
        const wrapper = shallow(<DecoratedToast {...defaultProps} title="Toast Title" />);
        expect(wrapper).toMatchSnapshot();
    });
});
