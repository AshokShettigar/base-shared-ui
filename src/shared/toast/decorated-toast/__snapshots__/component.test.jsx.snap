// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decorated Toast default renders 1`] = `
<InlineNotification
  isOpen={true}
  onClose={[Function]}
  type="message"
>
  <InlineNotificationMessage>
    Toast Message
  </InlineNotificationMessage>
</InlineNotification>
`;

exports[`Decorated Toast renders with props title 1`] = `
<InlineNotification
  isOpen={true}
  onClose={[Function]}
  type="message"
>
  <InlineNotificationTitle>
    Toast Title
  </InlineNotificationTitle>
  <InlineNotificationMessage>
    Toast Message
  </InlineNotificationMessage>
</InlineNotification>
`;
